### PR TITLE
Use poetry instead of pipenv

### PR DESCRIPTION
**Pull request by pehala**
* Poetry is in general a better tool, so migrating sooner than later is not a bad idea
  * It has better env management, you can have multiple pythons setup per project and Pycahrm respect them and doesn't create its own
  * It is quicker to sync resources
* Remove `pipenv` management from Makefile
     * This means the user is responsible for creating their own poetry env, there are helper `make` targets but they do not ensure it is up-to-date 
* Migrate all configuration files into `pyproject.toml`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -40,6 +40,7 @@ share/python-wheels/
 *.egg
 MANIFEST
 Pipfile.lock
```
Comment (jsmolar) [inline .gitignore:42]: Remove `Pipfile.lock`

**Review by jsmolar — DISMISSED** [review]
Approved, but do we still need `Pipfile.lock` in gitignore?

**Comment by pehala** [comment]
One additional change I made, I moved all configurations for our tools into `pyproject.toml`, which means we have less files we need to take care of.

**Review by JaurbanRH — APPROVED** [review]

**Comment by averevki** [comment]
I would like to also point out the format in which `poetry.lock` is saved. Instead of plain json by Pipenv, Poetry uses TOML format for its lock file. Which is more readable in my opinion.

**Review by jsmolar — DISMISSED** [review]

**Review by averevki — APPROVED** [review]

**Pull request merged**